src/coreclr/src/tools/aot/ILCompiler.TypeSystem.ReadyToRun.Tests/TestTypeSystemContext.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.TypeSystem;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using System.IO;
+
+namespace TypeSystemTests
+{
+    public enum CanonicalizationMode
+    {
+        Standard,
+        RuntimeDetermined,
+    }
+
+    class TestTypeSystemContext : MetadataTypeSystemContext
+    {
+        Dictionary<string, ModuleDesc> _modules = new Dictionary<string, ModuleDesc>(StringComparer.OrdinalIgnoreCase);
+
+        MetadataFieldLayoutAlgorithm _metadataFieldLayout = new TestMetadataFieldLayoutAlgorithm();
+        MetadataRuntimeInterfacesAlgorithm _metadataRuntimeInterfacesAlgorithm = new MetadataRuntimeInterfacesAlgorithm();
+#if !READYTORUN
+        ArrayOfTRuntimeInterfacesAlgorithm _arrayOfTRuntimeInterfacesAlgorithm;
+#endif
+        VirtualMethodAlgorithm _virtualMethodAlgorithm = new MetadataVirtualMethodAlgorithm();
+        
+        public CanonicalizationMode CanonMode { get; set; } = CanonicalizationMode.RuntimeDetermined;
+
+        public TestTypeSystemContext(TargetArchitecture arch)
+            : base(new TargetDetails(arch, TargetOS.Unknown, TargetAbi.Unknown))
+        {
+        }
+
+        public ModuleDesc GetModuleForSimpleName(string simpleName)
+        {
+            ModuleDesc existingModule;
+            if (_modules.TryGetValue(simpleName, out existingModule))
+                return existingModule;
+
+            return CreateModuleForSimpleName(simpleName);
+        }
+
+        public ModuleDesc CreateModuleForSimpleName(string simpleName)
+        {
+            string bindingDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            string filePath = Path.Combine(bindingDirectory, simpleName + ".dll");
+            ModuleDesc module = Internal.TypeSystem.Ecma.EcmaModule.Create(this, new PEReader(File.OpenRead(filePath)), containingAssembly: null);
+            _modules.Add(simpleName, module);
+            return module;
+        }
+
+        public override ModuleDesc ResolveAssembly(System.Reflection.AssemblyName name, bool throwIfNotFound)
+        {
+            return GetModuleForSimpleName(name.Name);
+        }
+
+        public override FieldLayoutAlgorithm GetLayoutAlgorithmForType(DefType type)
+        {
+#if !READYTORUN
```

Comment:
Do we need to if READYTORUN anything in this file now that the dependencies are ported over?

Reply:
Gonzo